src/py/flwr/common/auth_plugin/auth_plugin.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Abstract classes for Flower User Auth Plugin."""
+
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Any, Union
+
+import grpc
+
+from flwr.proto.exec_pb2 import (  # pylint: disable=E0611
+    GetAuthTokenRequest,
+    GetAuthTokenResponse,
+    LoginResponse,
+)
+from flwr.proto.exec_pb2_grpc import ExecStub
+
+
+class ExecAuthPlugin(ABC):
+    """Abstract Flower Auth Plugin class for ExecServicer."""
+
+    @abstractmethod
+    def __init__(self, config: dict[str, Any]):
+        """Abstract constructor (init)."""
+
+    @abstractmethod
+    def get_login_response(self) -> LoginResponse:
+        """Send relevant login details as a LoginResponse."""
+
+    @abstractmethod
+    def validate_token_in_metadata(
+        self, metadata: Sequence[tuple[str, Union[str, bytes]]]
+    ) -> bool:
+        """Authenticate auth tokens in the provided metadata."""
+
+    @abstractmethod
+    def get_auth_token_response(
+        self, request: GetAuthTokenRequest
+    ) -> GetAuthTokenResponse:
+        """Send relevant tokens as a GetAuthTokenResponse."""
```

Comment:
Is this method to communicate with the 3rd party service (like Keycloak) and return a `GetAuthTokenResponse`? In this case, should we amend the method as follows?
```suggestion
    def get_auth_tokens(
        self, request: GetAuthTokenRequest
    ) -> GetAuthTokenResponse:
        """Get the relevant auth tokens."""
```